hack/ci/e2e-k8s.sh

```diff
@@ -58,8 +58,15 @@ trap signal_handler INT TERM
 build() {
   # build the node image w/ kubernetes
   kind build node-image -v 1
+  # Ginkgo v1 is deprecated, perfer to build Ginkgo V2.
+  GINKGO_SRC_V2="vendor/github.com/onsi/ginkgo/v2/ginkgo"
+  if [ -d "$GINKGO_SRC_V2" ]; then
+      GINKGO_SRC_DIR="$GINKGO_SRC_V2"
+  else
+      GINKGO_SRC_DIR="vendor/github.com/onsi/ginkgo/ginkgo"
+  fi
```

Comment:
suggestion, we can do this more tersely with equivalent logic:

```sh
  # Ginkgo v1 is used by Kubernetes 1.24 and earlier, fallback if v2 is not available.
  GINKGO_SRC_DIR="vendor/github.com/onsi/ginkgo/v2/ginkgo"
  if [ ! -d "$GINKGO_SRC_DIR" ]; then
      GINKGO_SRC_DIR="vendor/github.com/onsi/ginkgo/ginkgo"
  fi
```

Reply:
Done.